### PR TITLE
Hide tiles on the moon

### PR DESCRIPTION
**Pull request by Argeento**
[https://github.com/bafolts/terraforming-mars/issues/3399](https://github.com/bafolts/terraforming-mars/issues/3399)

### CODE REVIEW

File: tests/components/overview/PlayerStatus.spec.ts [hunk 1]
```diff
@@ -1,34 +1,34 @@
-import {createLocalVue, mount} from '@vue/test-utils';
```
Comment (Argeento) [inline tests/components/overview/PlayerStatus.spec.ts:1]: These tests are commented because they don't pass

Comment (kberg) [reply tests/components/overview/PlayerStatus.spec.ts:1]: Thanks for catching the broken test and the bad format for the test parameters. Let's fix that first, separately. I'll pick on up that if you don't want to.

Comment (Argeento) [reply tests/components/overview/PlayerStatus.spec.ts:1]: Yes please. The code that this test was covering is now probably in `PlayerInfo` component.

Comment (kberg) [reply tests/components/overview/PlayerStatus.spec.ts:1]: Now in #3548.

File: src/components/Board.vue [hunk 2]
```diff
@@ -1,7 +1,9 @@
 <template>
     <div :class="getGameBoardClassName()">
         <div class="hide-tile-button-container">
-        <div class="hide-tile-button" v-on:click.prevent="toggleHideTile()">{{ toggleHideTileLabel() }}</div>
+          <div class="hide-tile-button" @click="$emit('toggleTileHidden')" data-test="hide-tiles-button">
```
Comment (kberg) [inline src/components/Board.vue:4]: I see -- data-test is the equivalent of an ID that doesn't have the semantics of a DOM id but can be used in tests? I like it. The phrase "data-test" isn't available online, so I wonder, is that just a local style you learned or a standard used elsewhere?

File: src/components/Board.vue [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+            {{ isTileHidden ? 'show' : 'hide' }} tiles
```
Comment (kberg) [inline src/components/Board.vue:5]: note to self: Ths isn't internationalized.

File: src/components/Board.vue [hunk 4]
```diff
@@ -56,7 +58,16 @@
...
         <div class="board" id="main_board">
-            <board-space :space="curSpace" :is_selectable="true" :key="'board-space-'+curSpace.id" :aresExtension="aresExtension" :isTileHidden="checkHideTile()" v-for="curSpace in getAllSpacesOnMars()"></board-space>
+            <board-space
+              v-for="curSpace in getAllSpacesOnMars()"
+              :key="curSpace.id"
+              :space="curSpace"
+              :is_selectable="true"
+              :aresExtension="aresExtension"
+              :isTileHidden="isTileHidden"
+              data-test="board-space"
```
Comment (kberg) [inline src/components/Board.vue:68]: Now I'm not sure what board-space represents, it's not an ID, it is however a semantic-free marker. So, that's helpful, too.

File: src/components/Board.vue (hunk 2 above)
Comment (kberg) [reply src/components/Board.vue:4]: Minor: `toggleTileHidden` -> `toggleHideTiles`

File: src/components/PlayerHome.vue [hunk 5]
```diff
@@ -256,6 +258,7 @@ export interface PlayerHomeModel {
   showActiveCards: boolean;
   showAutomatedCards: boolean;
   showEventCards: boolean;
+  isTileHidden: boolean;
```
Comment (kberg) [inline src/components/PlayerHome.vue:261]: 'is' is a standard for booleans, but here, I'd prefer `hideTiles`. Actually, I think everything should be expressed in their positive form (showTiles) regardless of their initial state, but that's not necessary here.

**Review by kberg — COMMENTED** [review]
This is really nice! I haven't finished reading it, but I like what I see, and I am learning alot. Thank you!

File: src/components/moon/MoonSpace.vue [hunk 6]
```diff
@@ -35,31 +40,45 @@ export default Vue.extend({
...
   },
-  methods: {
-    getVerboseTitle(tileType: TileType | undefined): string {
-      let ret: string = '';
-      if (tileType === TileType.LUNA_TRADE_STATION) {
+  computed: {
+    verboseTitle(): string {
+      let ret = '';
+      if (this.space.tileType === TileType.LUNA_TRADE_STATION) {
         ret = 'Luna Trade Station';
```
Comment (kberg) [inline src/components/moon/MoonSpace.vue:55]: Another reminder to internationlize these. Can you add a TODO for this and the spot above. please?

File: src/components/moon/MoonSpace.vue [hunk 7]
```diff
@@ -1,9 +1,14 @@
 <template>
-        <div :class="getMainClass()" :data_space_id="space.id" :title="getVerboseTitle(space.tileType)">
-            <div class="board-space-text" v-if="text" v-i18n>{{ text }}</div>
-            <bonus :bonus="space.bonus" v-if="space.tileType === undefined"></bonus>
-            <div :class="'board-cube board-cube--'+space.color" v-if="space.color !== undefined"></div>
-        </div>
+  <div :class="mainClass" :data_space_id="space.id" :title="verboseTitle">
```
Comment (kberg) [inline src/components/moon/MoonSpace.vue:2]: This is nice.

**Review by kberg — COMMENTED** [review]
This change is fantastic. If you can make the naming changes, it can go right in. Thank you!

File: src/components/Board.vue (hunk 2 above)
Comment (Argeento) [reply src/components/Board.vue:4]: It's often in use but with a different names - check `data-test-id` or `data-testid`. Some tools use their own conventions, like [`data-cy` in cypress](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements). But the idea is the same.

File: src/components/Board.vue (hunk 4 above)
Comment (Argeento) [reply src/components/Board.vue:68]: image
In test I had to somehow distinguish tiles on the moon from the around ones.

File: src/components/PlayerHome.vue (hunk 5 above)
Comment (Argeento) [reply src/components/PlayerHome.vue:261]: imo expression `doSomething` (`showTiles` in this case) is reserved for methods names

Comment (kberg) [reply src/components/PlayerHome.vue:261]: Understood. Verbs are used as properties (parameters) as well as methods. Thanks!

File: src/components/moon/MoonSpace.vue (hunk 6 above)
Comment (Argeento) [reply src/components/moon/MoonSpace.vue:56]: Done

File: src/components/Board.vue (hunk 3 above)
Comment (Argeento) [reply src/components/Board.vue:6]: Todo added

File: src/components/Board.vue (hunk 4 above)
Comment (kberg) [reply src/components/Board.vue:69]: Sorry, by board-space I meant data-test. But still, it's fine. :D

File: src/components/PlayerHome.vue (hunk 5 above)
Comment (kberg) [reply src/components/PlayerHome.vue:261]: Sorry perhaps I wasn't clear: I'm saying that hideTiles is an appropriate parameter name, and asking for the change.

**Review by kberg — COMMENTED** [review]
Thanks, just this last bit, then.

**Review by kberg — APPROVED** [review]
Thank you! borbarad will be thrilled.

**Pull request merged**